operators/src/source/gdal_source/loading_info.rs

```diff
@@ -0,0 +1,535 @@
+            params,
+            step,
+            dataset_time_start,
+            max_t2,
+            band_offset,
+        }
+    }
+
+    // fn band_offset(reference_time: TimeInstance, step: TimeStep, input_time: TimeInstance) -> usize {
+    //     reference_time + step * band_index as TimeInstance
+    // }
+}
+
+impl Iterator for NetCdfCfGdalLoadingInfoPartIterator {
+    type Item = Result<GdalLoadingInfoPart>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let t1 = self.time_step_iter.next()?;
+
+        // our first band is the reference time
+        // TODO: what if it intersects the reference time with t1 < ref?
+        if t1 < self.dataset_time_start || t1 >= self.max_t2 {
+            return None;
+        }
+
+        // snap t1 relative to reference time
+        // TODO: error handling instead of `ok`
+        let t1 = self.step.snap_relative(self.dataset_time_start, t1).ok()?;
+
+        let t2 = t1 + self.step;
+        let t2 = t2.unwrap_or(self.max_t2);
+
+        let time_interval = TimeInterval::new_unchecked(t1, t2);
+
+        // TODO: off by one if date is larger than reference time
+        let steps_between = if t1 == self.dataset_time_start {
+            0
+        } else {
+            1 + self
+                .step
+                .num_steps_in_interval(TimeInterval::new_unchecked(self.dataset_time_start, t1))
+                .unwrap() // TODO: what to do if this fails?
+        };
+
+        let mut params = self.params.clone();
+        params.rasterband_channel = self.band_offset + 1 + steps_between as usize;
+
+        // TODO: REMOVE
+        dbg!(
+            &params,
+            t1.as_rfc3339(),
+            self.dataset_time_start.as_rfc3339()
+        );
+
+        Some(Ok(GdalLoadingInfoPart {
+            time: time_interval,
+            params,
+        }))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct GdalLoadingInfo {
+    /// partitions of dataset sorted by time
+    pub info: GdalLoadingInfoPartIterator,
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Clone)]
+pub enum GdalLoadingInfoPartIterator {
+    Static {
+        parts: std::vec::IntoIter<GdalLoadingInfoPart>,
+    },
+    Dynamic(DynamicGdalLoadingInfoPartIterator),
+    NetCdfCf(NetCdfCfGdalLoadingInfoPartIterator),
+}
+
+impl Iterator for GdalLoadingInfoPartIterator {
+    type Item = Result<GdalLoadingInfoPart>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self {
+            GdalLoadingInfoPartIterator::Static { parts } => parts.next().map(Result::Ok),
+            GdalLoadingInfoPartIterator::Dynamic(iter) => iter.next(),
+            GdalLoadingInfoPartIterator::NetCdfCf(iter) => iter.next(),
+        }
+    }
+}
+
+/// one temporal slice of the dataset that requires reading from exactly one Gdal dataset
+#[derive(Debug, Clone, PartialEq)]
+pub struct GdalLoadingInfoPart {
+    pub time: TimeInterval,
+    pub params: GdalDatasetParameters,
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
We should  check if there are  tests for MEtaData in the "old" file and move them  here.

Reply:
8fc75bc